h11/_events.py

```diff
@@ -55,6 +58,10 @@ def __init__(self, **kwargs):
                 if not isinstance(self.status_code, int):
                     raise LocalProtocolError("status code must be integer")
 
+        if "target" in self.__dict__:
+            validate(request_target_re, self.__dict__["target"],
```

Comment:
You can just write `self.target` here, no need to use `__dict__`.

Also, this code should move to `Request._validate` – `_EventBundle.__init__` is where we put code that runs on multiple different event types, while fields that only show up in one event type get validated in that type's `_validate` method. (This is definitely confusing, sorry.)